### PR TITLE
[YUNIKORN-173] Generates one default application ID per namespace in the admission controller

### PR DESCRIPTION
**Pull request by yangwwei**
See more in https://issues.apache.org/jira/browse/YUNIKORN-173.

### CODE REVIEW

**Comment by yangwwei** [comment]
Tested locally, with or without namespace specified. the name generated will be like:

```
yunikorn-default-autogen
yunikorn-somenamespace-autogen
```

File: pkg/plugin/admissioncontrollers/webhook/admission_controller.go [hunk 1]
```diff
@@ -125,20 +129,23 @@ func updateSchedulerName(patch []patchOperation) []patchOperation {
...
 
-// the generated ID is using [PodName]_[Timestamp] naming convention.
-// the generated ID should be unique even the pod name is same (differ on nano time),
+// generate appID based on the namespace value,
 // and the max length of the ID is 63 chars.
-func generateAppID(podName string) string {
-	timestamp := strconv.FormatInt(time.Now().UnixNano(), 10)
-	generatedID := fmt.Sprintf("%.43s_%.19s", podName, timestamp)
-	return generatedID
+func generateAppID(namespace string) string {
```
Comment (wilfred-s) [inline pkg/plugin/admissioncontrollers/webhook/admission_controller.go:134]: Do we want to keep both options? One based on podName and the other based on namespace?
We have a config object and can then switch giving us the choice between namespace based and just on pod name. Seems a waste to throw away that option.

File: pkg/plugin/admissioncontrollers/webhook/admission_controller.go [hunk 2]
```diff
@@ -149,16 +156,9 @@ func updateLabels(pod *v1.Pod, patch []patchOperation) []patchOperation {
...
-			podName := "unknown"
-			if pod.Name != "" {
-				podName = pod.Name
-			} else if pod.GenerateName != "" {
-				podName = pod.GenerateName
-			}
-
-			generatedID := generateAppID(podName)
+			// for each namespace, we group unnamed pods to one single app
+			// application ID convention: __${NAMESPACE}_app_default
```
Comment (wilfred-s) [inline pkg/plugin/admissioncontrollers/webhook/admission_controller.go:160]: This does not line up with the test and what we generate a `prefix`-${NAMESPACE}-`suffix`

Comment (yangwwei) [reply pkg/plugin/admissioncontrollers/webhook/admission_controller.go]: Fixed

File: pkg/plugin/admissioncontrollers/webhook/admission_controller.go (hunk 1 above)
Comment (yangwwei) [reply pkg/plugin/admissioncontrollers/webhook/admission_controller.go:134]: That will be a bit hard to accomplish. We don't have an option to config the admission controller's behavior today.

**Pull request closed without merging**